Tune dependency scopes

jjohannes opened the pull request:
**Description**:

Correct everything suggested by `./gradlew checkAllModuleInfo`. In particular, this reverts #8347, which was a remedy to get out of the Gradle deadlock bug, which should now be fixed for good by #8352

<!--
One or two line summary of what this PR does and why it is needed, followed by a list
of changes in imperative, present tense for use in the commit message or changelog. Example:

This PR modifies ... in order to support ...
* Add config property
* Change column name
* Remove ...
-->

**Related issue(s)**:

**Notes for reviewer**:
<!-- Provide logs, performance numbers or screenshots of the new functionality -->

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

nathanklick commented:
@jjohannes The dependency check plugin is stuck on `require org.bouncycastle.provider` in the `test-clients` module even though the `module-info.java` has the exact require statement it is wanting. 

Please take a look at this failure when you return.

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test-clients:checkModuleDirectivesScope'.
> /runner/_work/hedera-services/hedera-services/hedera-node/test-clients/src/main/java/module-info.java
  
  Please add the following requires directives:
      requires org.bouncycastle.provider;

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org./

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
2969 actionable tasks: 1929 executed, 805 from cache, 235 up-to-date

BUILD FAILED in 41s

Publishing build scan...
https://gradle.com/s/kyrliktcqudrk
```

jjohannes commented:
@nathanklick we are hitting this issue with the dependency analysis plugin: https://github.com/autonomousapps/dependency-analysis-gradle-plugin/issues/947

Different versions of `bcprov` are used for production vs. test time. – `jdk15on` vs `jdk18on`. I don't know why we use `15on` (Java 5 compatible) and not `jdk18on` (Java 8 compatible) at all. I already changed that it #8561. I now moved this change to this PR to resolve the problem here.

I also rebased latest develop and addressed what the check reported after that.
